dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityQueryParams.java

```diff
@@ -196,19 +190,19 @@ public TrackedEntityQueryParams addOrganisationUnit(OrganisationUnit unit) {
    */
   public void handleOrganisationUnits() {
```

Comment:
Is there a way to get rid of this method? Does not have to be in this PR of course 😅 

When working on `order` there was something similar in I think it was events query params. The reason it existed was a poor API design. If some fields/collections need to be kept in sync or in a correct state space then the class itself should take care of that. We should not require clients to know that you need to initialize the class, set what you want and then call some `handleYYYY` method that then makes sure the fields are in a correct state.

For events you needed to know that if you want to order by an attribute you also add it to the attributes and filters collection of EventQueryParams 😓 Things like that need to be internal and happen on calls to in that case `orderBy(tea,asc)`.

Here it might be on adding an orgUnit 🤷🏻

Reply:
The problem I have with this is that I want to break down this ticket into multiple PRs. 
I started with this one where I refactor the service, and to make sure it works as expected, I need to leave this ugly piece of code here. When I start working on the store, I'll remove it.